src/graphql/public/root/mutation/account-update-push-notification-settings.ts

```diff
@@ -0,0 +1,56 @@
+import { GT } from "@graphql/index"
+
+import PushNotificationType from "@graphql/shared/types/scalar/push-notification-type"
+import AccountUpdatePushNotificationSettingsPayload from "@graphql/public/types/payload/account-update-push-notification-settings"
+import { Accounts } from "@app/index"
+import { mapAndParseErrorForGqlResponse } from "@graphql/error-map"
+
+const AccountUpdatePushNotificationSettingsInput = GT.Input({
+  name: "AccountUpdatePushNotificationSettingsInput",
+  fields: () => ({
+    pushNotificationsEnabled: {
+      type: GT.NonNull(GT.Boolean),
+    },
+    disabledPushNotificationTypes: {
+      type: GT.NonNull(GT.List(PushNotificationType)),
+    },
```

Comment:
I still think this should be segmented into 2 mutations. 

1 mutation = 1 action. 

here both argument are mandatory. so action such as:

"I want to deactivate one notification": it is unclear whether the `pushNotificationsEnabled` should be set to true or false.

Reply:
I think it depends on how you view the `1 action`. Going to the notification settings page and then turning on or off a few notifications and then sending your new notification settings when you leave the page could be seen as one action "updating your setting". Otherwise you could look at each individual toggle as 1 action. I'm open to either one. If you split this into two queries, would you then add two more queries for enabling and disabling email and selecting notifications that are disabled in email?